packages/orchestration/src/exos/chain-hub.js

```diff
@@ -133,19 +128,26 @@ export const makeChainHub = (agoricNames, zone = makeHeapZone()) => {
       // Either from registerChain or memoized remote lookup()
       if (chainInfos.has(chainName)) {
         return /** @type {Vow<ActualChainInfo<K>>} */ (
-          watch(chainInfos.get(chainName))
+          vowTools.asVow(() => chainInfos.get(chainName))
         );
       }
 
-      return watch(E(agoricNames).lookup(CHAIN_KEY, chainName), {
-        onFulfilled: chainInfo => {
-          chainInfos.init(chainName, chainInfo);
-          return chainInfo;
-        },
-        onRejected: _cause => {
-          throw makeError(`chain not found:${chainName}`);
+      const lookup = vowTools.retriable(
+        zone,
+        'getConnectionInfo',
+        // eslint-disable-next-line no-restricted-syntax -- TODO more exact rules for vow best practices
+        async () => {
+          await null;
+          try {
+            const chainInfo = await E(agoricNames).lookup(CHAIN_KEY, chainName);
+            chainInfos.init(chainName, chainInfo);
+            return chainInfo;
+          } catch (_) {
+            throw makeError(`chain not found:${chainName}`);
+          }
         },
-      });
```

Comment:
This needs to be defined not nested in a function call as it's a prepare pattern. Sorry that's a fidelity issue with the current stub